src/Client/Provider/GithubClient.php

```diff
@@ -22,7 +22,29 @@ class GithubClient extends OAuth2Client
      */
     public function fetchUserFromToken(AccessToken $accessToken)
     {
-        return parent::fetchUserFromToken($accessToken);
+        $resourceOwner = parent::fetchUserFromToken($accessToken);
+
+        if (null === $resourceOwner->getEmail()) {
+            $response = $this->getOAuth2Provider()->getHttpClient()->request(
+                'GET',
+                'https://api.github.com/user/emails',
+                [
+                    'headers' => [
+                        'Authorization' => "Bearer {$accessToken->getToken()}"
+                    ]
+                ]
+            );
+
+            foreach (json_decode($response->getBody()->getContents(), true) as $email) {
+                if (true === $email['primary']) {
```

Comment:
Are we sure `primary` key always exists on `$email`?  I mean, probably we need to do `isset($email['primary'])` first?

Reply:
According to the [official documentation](https://developer.github.com/v3/users/emails/) and my personal tests, yes this key is always present.